compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1051,30 +1050,25 @@ public void visit(BLangFieldBasedAccess fieldAccessExpr) {
         }
 
         BLangVariableReference targetVarRef = fieldAccessExpr;
-        if (fieldAccessExpr.expr.type.tag == TypeTags.ENUM) {
-            targetVarRef = new BLangEnumeratorAccessExpr(fieldAccessExpr.pos,
-                    fieldAccessExpr.field, (BVarSymbol) fieldAccessExpr.symbol);
-        } else {
-            fieldAccessExpr.expr = rewriteExpr(fieldAccessExpr.expr);
-            BLangLiteral stringLit = createStringLiteral(fieldAccessExpr.pos, fieldAccessExpr.field.value);
-            BType varRefType = fieldAccessExpr.expr.type;
-            if (varRefType.tag == TypeTags.OBJECT || varRefType.tag == TypeTags.RECORD) {
-                if (fieldAccessExpr.symbol instanceof BInvokableSymbol &&
-                        ((fieldAccessExpr.symbol.flags & Flags.ATTACHED) == Flags.ATTACHED)) {
-                    targetVarRef = new BLangStructFunctionVarRef(fieldAccessExpr.expr,
-                                                                 (BVarSymbol) fieldAccessExpr.symbol);
-                } else {
-                    targetVarRef = new BLangStructFieldAccessExpr(fieldAccessExpr.pos, fieldAccessExpr.expr, stringLit,
-                                                                  (BVarSymbol) fieldAccessExpr.symbol);
-                }
-            } else if (varRefType.tag == TypeTags.MAP) {
-                targetVarRef = new BLangMapAccessExpr(fieldAccessExpr.pos, fieldAccessExpr.expr, stringLit);
-            } else if (varRefType.tag == TypeTags.JSON) {
-                targetVarRef = new BLangJSONAccessExpr(fieldAccessExpr.pos, fieldAccessExpr.expr, stringLit);
-            } else if (varRefType.tag == TypeTags.XML) {
-                targetVarRef = new BLangXMLAccessExpr(fieldAccessExpr.pos, fieldAccessExpr.expr, stringLit,
-                                                      fieldAccessExpr.fieldKind);
+        fieldAccessExpr.expr = rewriteExpr(fieldAccessExpr.expr);
+        BLangLiteral stringLit = createStringLiteral(fieldAccessExpr.pos, fieldAccessExpr.field.value);
+        BType varRefType = fieldAccessExpr.expr.type;
+        if (varRefType.tag == TypeTags.OBJECT || varRefType.tag == TypeTags.RECORD) {
+            if (fieldAccessExpr.symbol instanceof BInvokableSymbol &&
```

Comment:
Shall we use fieldAccessExpr.symbol.getType().tag == TypeTags.INVOKABLE instead of instanceof here

Reply:
Even symbol tag can differ, type tag should be invokable tag. Can you please check?

Reply:
Yes, fixed with 27da939, thanks!